distributed/scheduler.py

```diff
@@ -6500,6 +6500,41 @@ def get_events(self, comm=None, topic=None):
         else:
             return valmap(tuple, self.events)
 
+    async def get_worker_monitor_info(self, starts=dict(), recent=False):
+        parent: SchedulerState = cast(SchedulerState, self)
+        query = asyncio.gather(
+            *(
+                self.rpc(w).get_monitor_range(start=starts.get(w, 0))
+                for w in parent._workers_dv
+            )
+        )
+        if recent:
+            query = asyncio.gather(
+                *(self.rpc(w).get_monitor_recent() for w in parent._workers_dv)
+            )
+        return {
```

Comment:
Most scheduler operations that return information for each worker tend to return dictionaries with worker addresses for the keys and the corresponding values are the information the each worker (see, for example, [this method](https://github.com/dask/distributed/blob/b5f3fff0a03df157451adffcf8e3a8a4cee88199/distributed/scheduler.py#L5925-L5940) and the methods around it). It'd be good to keep the same format here. We can have this method return a dictionary like 

```
{
<worker-1-address>: {"count": ..., "range_query": ..., ...},
<worker-2-address>: {"count": ..., "range_query": ..., ...},
<worker-3-address>: {"count": ..., "range_query": ..., ...},
...
}
```

Additionally, if the worker only had a single `Worker.get_monitor_info` method, we could use [`Scheduler.broadcast`](https://github.com/dask/distributed/blob/b5f3fff0a03df157451adffcf8e3a8a4cee88199/distributed/scheduler.py#L5198) to get this return value in a straightforward way (see [here](https://github.com/dask/distributed/blob/b5f3fff0a03df157451adffcf8e3a8a4cee88199/distributed/scheduler.py#L6097-L6104) for a similar example where we use `Scheduler.broadcast`)

Reply:
Relating to the above conversation, can you supply per-worker arguments to `broadcast()` (such as the per-worker `start` values)?

Reply:
Yes, a call like 

```python
await self.rpc(w).funcname(key=value)
```

Actually sends the following message

```python
{"op": "funcname", "key: "value"}
```

So you should be able to specify keyword arguments by crafting the appropriate value for the `msg=` keyword of broadcast.

Reply:
I understand that you can supply arguments to the workers in that way, but it seems like in that case all workers would receive `key=value`. Is there a way to supply different arguments to each worker, such as:

```python
{
    "op": "funcname", 
    "key": {
        "worker-1": arg_1,
        "worker-2": arg_2,
        ...
    }
}
```

This would be to supply worker-unique values for `start`, which I'm [not necessarily sure is needed](https://github.com/dask/distributed/pull/4657#discussion_r605948271).